src/pages/en/core-concepts/astro-components.md

```diff
@@ -101,6 +101,206 @@ const myFavoritePokemon = [/* ... */];
 
 ```
 
+### Dynamic JSX Expressions
+
+Astro components can define local variables inside of the frontmatter component script. Any script variables are then automatically available in the component's HTML template below.
+
+#### Dynamic Values
+
+These local variables can be used in curly braces to pass values to be used as HTML output:
+
+```astro
+---
+const name = "Astro";
+---
+<div>
+  <h1>Hello {name}!</h1>
+</div>
+```
+
+#### Dynamic Attributes
+
+These local variables can be used in curly braces to pass attribute values to HTML elements and components:
+
+```astro
+---
+const name = "Astro";
+---
+<h1 class={name}>Attribute expressions are supported</h1>
+
+<MyComponent templateLiteralNameAttribute={`MyNameIs${name}`} />
+```
+
+#### Dynamic HTML
+
+These local variables can be used in JSX-like functions to produce dynamically-generated HTML elements:
+
+```astro
+---
+const items = ["Dog", "Cat", "Platipus"];
+---
+<ul>
+  {items.map((item) => (
+    <li>{item}</li>
+  ))}
+</ul>
+```
+
+#### Fragments & Multiple Elements
+
+Remember: an Astro component template can render multiple elements with no need to wrap everything in a single `<div>` or `<>`.
+
+However, when using an Astro JSX-like expression to dynamically create elements, you must wrap these multiple elements inside of a **Fragment** just like you would in JavaScript or JSX. Astro supports using either `<Fragment> </Fragment>` or `<> </>`.
+
+```astro
+---
+const items = ["Dog", "Cat", "Platipus"];
+---
+<ul>
+  {items.map((item) => (
+    <>
+      <li>Red {item}</li>
+      <li>Blue {item}</li>
+      <li>Green {item}</li>
+    </>
+  ))}
+</ul>
+```
+
+
+### Component Props
+
+An Astro component can define and accept props. These props then become available to the component template for rendering HTML. Props are available on the `Astro.props` global in your frontmatter script.
+
+Here is an example of a component that receives a `greeting` prop and a `name` prop. Notice that the props to be received are destructured from the global `Astro.props` object.
+
+```astro
+// src/components/GreetingHeadline.astro
```

Comment:
this comments should always be inside the `---` code fence, so that the code is valid. Otherwise syntax highlighting can get very odd.

Reply:
also, can you show an example of how this component is used? Maybe something like:


```astro
---
// Example: GreetingHeadline.astro
// Usage: <GreetingHeadline greeting="Howdy" name="Partner" />
const { greeting, name } = Astro.props
---
<h2>{greeting}, {name}!</h2>
```

Reply:
Correct `// comments` can only go inside of the `---` fence.